data_deletion/__init__.py

```diff
@@ -1,30 +1,52 @@
 from datetime import datetime
-from os import listdir
-from os.path import join
-from egcg_core import app_logging, executor
+from os import listdir, stat
+from os.path import join, isdir
+
+from cached_property import cached_property
+
+from egcg_core import app_logging, executor, clarity, rest_communication, util
+from egcg_core.app_logging import AppLogger
+from egcg_core.archive_management import is_archived, ArchivingError
+from egcg_core.config import cfg
+from egcg_core.constants import ELEMENT_SAMPLE_INTERNAL_ID, ELEMENT_PROJECT_ID, ELEMENT_SAMPLE_EXTERNAL_ID, \
+    ELEMENT_RUN_NAME, ELEMENT_LANE
 from egcg_core.exceptions import EGCGError
 
 
+def get_file_list_size(file_list):
+    """
+    Get the total size of all files. Collapses them by inodes to avoid counting hard links more than once.
+    Also descends into directories recursively.
+    """
+    def files_by_inode(_file_list):
+        _inode2file = {}
+        for f in _file_list:
+            if isdir(f):
+                _inode2file.update(files_by_inode(join(f, s) for s in listdir(f)))
+            else:
+                _inode2file[stat(f).st_ino] = f
+        return _inode2file
+
+    return sum(stat(f).st_size for f in files_by_inode(file_list).values())
+
+
 class DataDeletionError(EGCGError):
     pass
 
 
 class Deleter(app_logging.AppLogger):
     data_dir = ''
     local_execute_only = False
-    _deletion_dir = None
 
     def __init__(self, work_dir, dry_run=False, deletion_limit=None):
         self.work_dir = work_dir
         self.dry_run = dry_run
         self.deletion_limit = deletion_limit
 
-    @property
+    @cached_property
```

Comment:
cached_property could probably help clean up quite a lot of our other projects as well